### PR TITLE
Use async/await for extension functions

### PR DESCRIPTION
**Pull request by darkwing**
Trying to modernize some of our code; the await's are so much easier to read.

### CODE REVIEW

**Review by Gudahtt — DISMISSED** [review]
LGTM!

**Comment by jaywilliamsapar** [comment]
Nice

File: app/scripts/platforms/extension.js [hunk 1]
```diff
@@ -13,70 +13,52 @@ export default class ExtensionPlatform {
...
-        const error = checkForError();
-        if (error) {
-          return reject(error);
-        }
-        return resolve(newTab);
-      });
-    });
+  async openTab(options) {
+    const newTab = await browser.tabs.create(options);
+    const error = checkForError();
```
Comment (Gudahtt) [inline app/scripts/platforms/extension.js:18]: I don't think we need the `checkForError` steps anymore when using the Promise-based APIs.
See: https://github.com/mozilla/webextension-polyfill#using-the-promise-based-apis

Comment (darkwing) [reply app/scripts/platforms/extension.js]: Nice!

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by brad-decker — APPROVED** [review]

**Pull request merged**